### PR TITLE
update weights, attempt two (0.9.31)

### PR DESCRIPTION
**Pull request by coderobe**
https://github.com/paritytech/polkadot/pull/6173 regenerated with https://github.com/paritytech/substrate/pull/12482 applied

### CODE REVIEW

File: runtime/kusama/src/weights/pallet_conviction_voting.rs [hunk 1]
```diff
@@ -49,34 +48,38 @@ impl<T: frame_system::Config> pallet_conviction_voting::WeightInfo for WeightInf
...
 	// Storage: ConvictionVoting ClassLocksFor (r:1 w:1)
 	// Storage: Balances Locks (r:1 w:1)
-	// Storage: Scheduler Agenda (r:2 w:2)
+	// Storage: Scheduler Agenda (r:1 w:1)
 	fn vote_new() -> Weight {
-		Weight::from_ref_time(5_652_326_000 as u64)
-			.saturating_add(T::DbWeight::get().reads(6 as u64))
-			.saturating_add(T::DbWeight::get().writes(6 as u64))
+		// Minimum execution time: 125_577 nanoseconds.
+		Weight::from_ref_time(128_422_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_conviction_voting.rs:54]: big jump down

File: runtime/kusama/src/weights/pallet_conviction_voting.rs [hunk 2]
```diff
@@ -87,32 +90,35 @@ impl<T: frame_system::Config> pallet_conviction_voting::WeightInfo for WeightInf
 	// Storage: Scheduler Agenda (r:2 w:2)
 	/// The range of component `r` is `[0, 512]`.
 	fn delegate(r: u32, ) -> Weight {
-		Weight::from_ref_time(78_376_000 as u64)
-			// Standard Error: 2_253_708
-			.saturating_add(Weight::from_ref_time(221_428_037 as u64).saturating_mul(r as u64))
-			.saturating_add(T::DbWeight::get().reads(4 as u64))
+		// Minimum execution time: 76_733 nanoseconds.
+		Weight::from_ref_time(707_216_284 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_conviction_voting.rs:94]: x10 jump up

File: runtime/kusama/src/weights/pallet_conviction_voting.rs [hunk 3]
```diff
@@ -87,32 +90,35 @@ impl<T: frame_system::Config> pallet_conviction_voting::WeightInfo for WeightInf
...
 	// Storage: Referenda ReferendumInfoFor (r:10 w:10)
 	// Storage: Scheduler Agenda (r:2 w:2)
 	/// The range of component `r` is `[0, 512]`.
 	fn undelegate(r: u32, ) -> Weight {
-		Weight::from_ref_time(55_946_000 as u64)
-			// Standard Error: 2_268_396
-			.saturating_add(Weight::from_ref_time(221_648_859 as u64).saturating_mul(r as u64))
-			.saturating_add(T::DbWeight::get().reads(2 as u64))
+		// Minimum execution time: 52_748 nanoseconds.
+		Weight::from_ref_time(683_979_746 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_conviction_voting.rs:108]: x10 jump up

File: runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs [hunk 4]
```diff
@@ -139,16 +143,11 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(874_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 14_000
-			.saturating_add(Weight::from_ref_time(39_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 23_000
-			.saturating_add(Weight::from_ref_time(10_873_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 35_000
-			.saturating_add(Weight::from_ref_time(2_257_000 as u64).saturating_mul(d as u64))
+	fn submit_unsigned(_v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 6_771_975 nanoseconds.
+		Weight::from_ref_time(6_832_732_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs:148]: big jump up

File: runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs [hunk 5]
```diff
@@ -160,16 +159,13 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			// Standard Error: 25_000
-			.saturating_add(Weight::from_ref_time(18_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 42_000
-			.saturating_add(Weight::from_ref_time(8_673_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 63_000
-			.saturating_add(Weight::from_ref_time(1_598_000 as u64).saturating_mul(d as u64))
+	fn feasibility_check(v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 5_640_118 nanoseconds.
+		Weight::from_ref_time(5_681_788_000 as u64)
+			// Standard Error: 18_489
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs:165]: big jump up

File: runtime/kusama/src/weights/pallet_elections_phragmen.rs [hunk 6]
```diff
@@ -148,13 +158,12 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
...
-	fn clean_defunct_voters(v: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 88_000
-			.saturating_add(Weight::from_ref_time(60_894_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 88_000
-			.saturating_add(Weight::from_ref_time(379_000 as u64).saturating_mul(d as u64))
+	/// The range of component `d` is `[0, 5000]`.
+	fn clean_defunct_voters(v: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 277_672_844 nanoseconds.
+		Weight::from_ref_time(278_514_899_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_elections_phragmen.rs:164]: huge jump up

File: runtime/kusama/src/weights/pallet_elections_phragmen.rs [hunk 7]
```diff
@@ -172,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
 	/// The range of component `v` is `[1, 10000]`.
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 691_000
-			.saturating_add(Weight::from_ref_time(57_805_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 46_000
-			.saturating_add(Weight::from_ref_time(3_139_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 28_457_507 nanoseconds.
+		Weight::from_ref_time(28_618_929_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_elections_phragmen.rs:185]: one more huge jump up

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 8]
```diff
@@ -332,21 +353,21 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
+	/// The range of component `n` is `[0, 100]`.
 	fn new_era(v: u32, n: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 579_000
-			.saturating_add(Weight::from_ref_time(161_445_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 55_000
-			.saturating_add(Weight::from_ref_time(22_905_000 as u64).saturating_mul(n as u64))
-			.saturating_add(T::DbWeight::get().reads(192 as u64))
+		// Minimum execution time: 441_741 nanoseconds.
+		Weight::from_ref_time(444_325_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_staking.rs:365]: big jump up

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 9]
```diff
@@ -365,27 +386,28 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
 	fn get_npos_voters(v: u32, n: u32, s: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 375_000
-			.saturating_add(Weight::from_ref_time(41_603_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 375_000
-			.saturating_add(Weight::from_ref_time(38_528_000 as u64).saturating_mul(n as u64))
-			// Standard Error: 9_578_000
-			.saturating_add(Weight::from_ref_time(41_537_000 as u64).saturating_mul(s as u64))
+		// Minimum execution time: 25_700_092 nanoseconds.
+		Weight::from_ref_time(25_792_272_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_staking.rs:390]: Big jump up

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 10]
```diff
@@ -365,27 +386,28 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
 	// Storage: Staking Validators (r:501 w:0)
 	// Storage: System BlockWeight (r:1 w:1)
 	/// The range of component `v` is `[500, 1000]`.
 	fn get_npos_targets(v: u32, ) -> Weight {
-		Weight::from_ref_time(310_326_000 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(6_164_000 as u64).saturating_mul(v as u64))
-			.saturating_add(T::DbWeight::get().reads(2 as u64))
+		// Minimum execution time: 3_338_127 nanoseconds.
+		Weight::from_ref_time(3_386_919_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_staking.rs:407]: x10 jump up

File: runtime/kusama/src/weights/runtime_parachains_hrmp.rs [hunk 11]
```diff
@@ -87,11 +90,12 @@ impl<T: frame_system::Config> runtime_parachains::hrmp::WeightInfo for WeightInf
 	/// The range of component `i` is `[0, 127]`.
 	/// The range of component `e` is `[0, 127]`.
 	fn force_clean_hrmp(i: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 15_000
-			.saturating_add(Weight::from_ref_time(7_134_000 as u64).saturating_mul(i as u64))
-			// Standard Error: 15_000
-			.saturating_add(Weight::from_ref_time(7_238_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 856_131 nanoseconds.
+		Weight::from_ref_time(862_826_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/runtime_parachains_hrmp.rs:94]: Big jump up

File: runtime/kusama/src/weights/frame_election_provider_support.rs [hunk 12]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmen(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(13_944_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 4_876_000
-			.saturating_add(Weight::from_ref_time(2_223_649_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 5_442_855 nanoseconds.
+		Weight::from_ref_time(5_493_581_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/frame_election_provider_support.rs:52]: big jump up from 0

File: runtime/kusama/src/weights/frame_election_provider_support.rs [hunk 13]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmms(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 79_000
-			.saturating_add(Weight::from_ref_time(14_480_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 6_844_000
-			.saturating_add(Weight::from_ref_time(2_525_332_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 4_312_432 nanoseconds.
+		Weight::from_ref_time(4_352_242_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/frame_election_provider_support.rs:63]: here as well

File: runtime/polkadot/src/weights/frame_election_provider_support.rs [hunk 14]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmen(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(13_900_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 4_843_000
-			.saturating_add(Weight::from_ref_time(2_216_812_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 5_430_677 nanoseconds.
+		Weight::from_ref_time(5_464_877_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/frame_election_provider_support.rs:52]: big jump up from 0

File: runtime/polkadot/src/weights/frame_election_provider_support.rs [hunk 15]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmms(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 84_000
-			.saturating_add(Weight::from_ref_time(14_955_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 7_259_000
-			.saturating_add(Weight::from_ref_time(2_615_111_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 4_281_157 nanoseconds.
+		Weight::from_ref_time(4_309_020_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/frame_election_provider_support.rs:63]: same here

File: runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs [hunk 16]
```diff
@@ -158,16 +163,13 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(789_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 8_000
-			.saturating_add(Weight::from_ref_time(41_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 13_000
-			.saturating_add(Weight::from_ref_time(5_713_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 20_000
-			.saturating_add(Weight::from_ref_time(1_311_000 as u64).saturating_mul(d as u64))
+	fn feasibility_check(v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 3_977_796 nanoseconds.
+		Weight::from_ref_time(3_993_895_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs:168]: big jump up from 0

File: runtime/polkadot/src/weights/pallet_elections_phragmen.rs [hunk 17]
```diff
@@ -172,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
 	/// The range of component `v` is `[1, 10000]`.
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 674_000
-			.saturating_add(Weight::from_ref_time(52_473_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 44_000
-			.saturating_add(Weight::from_ref_time(2_826_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 27_528_355 nanoseconds.
+		Weight::from_ref_time(27_615_108_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_elections_phragmen.rs:185]: big jump up from 0

File: runtime/polkadot/src/weights/pallet_staking.rs [hunk 18]
```diff
@@ -365,27 +384,28 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 450_000
-			.saturating_add(Weight::from_ref_time(38_224_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 450_000
-			.saturating_add(Weight::from_ref_time(38_981_000 as u64).saturating_mul(n as u64))
-			// Standard Error: 11_498_000
-			.saturating_add(Weight::from_ref_time(16_085_000 as u64).saturating_mul(s as u64))
+		// Minimum execution time: 25_507_250 nanoseconds.
+		Weight::from_ref_time(25_649_540_000 as u64)
+			// Standard Error: 501_634
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_staking.rs:389]: big jump up from 0

File: runtime/rococo/src/weights/pallet_elections_phragmen.rs [hunk 19]
```diff
@@ -148,11 +158,12 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
...
 	// Storage: System Account (r:5000 w:5000)
 	/// The range of component `v` is `[5000, 10000]`.
-	/// The range of component `d` is `[1, 5000]`.
+	/// The range of component `d` is `[0, 5000]`.
 	fn clean_defunct_voters(v: u32, _d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 85_000
-			.saturating_add(Weight::from_ref_time(61_507_000 as u64).saturating_mul(v as u64))
+		// Minimum execution time: 278_626_833 nanoseconds.
+		Weight::from_ref_time(279_736_954_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/rococo/src/weights/pallet_elections_phragmen.rs:164]: big jump up from 0

File: runtime/rococo/src/weights/pallet_elections_phragmen.rs [hunk 20]
```diff
@@ -170,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
 	/// The range of component `v` is `[1, 10000]`.
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 1_864_000
-			.saturating_add(Weight::from_ref_time(167_385_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 124_000
-			.saturating_add(Weight::from_ref_time(9_721_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 28_441_420 nanoseconds.
+		Weight::from_ref_time(28_674_262_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/rococo/src/weights/pallet_elections_phragmen.rs:185]: big jump up from 0

File: runtime/westend/src/weights/frame_election_provider_support.rs [hunk 21]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmen(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 58_000
-			.saturating_add(Weight::from_ref_time(14_001_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 5_013_000
-			.saturating_add(Weight::from_ref_time(2_245_454_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 5_449_895 nanoseconds.
+		Weight::from_ref_time(5_493_777_000 as u64)
+			// Standard Error: 137_519
```
Comment (EgorPopelyaev) [inline runtime/westend/src/weights/frame_election_provider_support.rs:53]: big jump up from 0

File: runtime/westend/src/weights/frame_election_provider_support.rs [hunk 22]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmms(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 84_000
-			.saturating_add(Weight::from_ref_time(15_041_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 7_315_000
-			.saturating_add(Weight::from_ref_time(2_619_056_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 4_267_232 nanoseconds.
+		Weight::from_ref_time(4_311_499_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/westend/src/weights/frame_election_provider_support.rs:63]: big jump up from 0

File: runtime/westend/src/weights/pallet_staking.rs [hunk 23]
```diff
@@ -365,27 +386,28 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
 	fn get_npos_voters(v: u32, n: u32, s: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 378_000
-			.saturating_add(Weight::from_ref_time(36_021_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 378_000
-			.saturating_add(Weight::from_ref_time(34_834_000 as u64).saturating_mul(n as u64))
-			// Standard Error: 9_659_000
-			.saturating_add(Weight::from_ref_time(52_904_000 as u64).saturating_mul(s as u64))
+		// Minimum execution time: 25_695_485 nanoseconds.
+		Weight::from_ref_time(25_860_451_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/westend/src/weights/pallet_staking.rs:390]: big jump up from 0

**Comment by ggwpez** [comment]
The only case where there are a large number of storage ops is here (which is on master for Kusama, so should be fine?):  
image
cc @koute @shawntabrizi

**Comment by ggwpez** [comment]
Okay I think the ~3600 READ/WRITE is fine in on_init and is already live for a long time on the other runtimes.  
On rococo it now happened since https://github.com/paritytech/polkadot/pull/5617 did not update the weights for the auctions.

**Review by ggwpez — APPROVED** [review]
Looks like `runtime/rococo/src/weights/runtime_parachains_paras_inherent.rs` was not updated.  
No idea why. Maybe missing from a script?  
In fact it was never updated in the release process so far.

**Comment by ggwpez** [comment]
Ah the para-inherents for rococo does not work. I remember there being an issue. Should be fine :smile:

**Pull request merged**